src/app/shared/components/dynamic-form/dynamic-form.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Validator } from './../../interfaces/field.interface';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  ViewChild,
+  ComponentFactoryResolver,
+  ViewContainerRef
+} from '@angular/core';
+import {
+  FormGroup,
+  FormBuilder,
+  Validators,
+} from '@angular/forms';
+import { FieldConfig } from '@shared/interfaces/field.interface';
+
+@Component({
+  selector: 'app-dynamic-form',
+  templateUrl: './dynamic-form.component.html',
+  styles: []
+})
+export class DynamicFormComponent implements OnInit  {
+  @Input() fields: FieldConfig[] = [];
+
+  @Output() save: EventEmitter<any> = new EventEmitter<any>();
+  @Output() emitHandler: EventEmitter<any> = new EventEmitter<any>();
+
+  form: FormGroup;
+
+  get value() {
+    return this.form.value;
+  }
+
+  constructor(
+    private fb: FormBuilder,
+    public resolver: ComponentFactoryResolver,
+    public container: ViewContainerRef
+  ) {
+  }
+
+  ngOnInit() {
+    this.form = this.createControl();
+  }
+
+  onSubmit(event: Event) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (this.form.valid) {
+      this.save.emit(this.form.value);
+    } else {
+      this.validateAllFormFields(this.form);
+    }
+  }
+
+  private createControl(): FormGroup {
+    const group = this.fb.group({});
+    this.fields.forEach((field: FieldConfig) => {
+      if (field.type === 'button') {
+        return;
+      }
+      const control = this.fb.control(
```

Comment:
type

Reply:
not sure in this case, because `this.fb.control` - already indicates type